proxy/logging/LogField.h

```diff
@@ -82,6 +82,12 @@ class LogField
   typedef int (*UnmarshalFuncWithMap)(char **buf, char *dest, int len, const Ptr<LogFieldAliasMap> &map);
   typedef void (LogAccess::*SetFunc)(char *buf, int len);
 
+  static UnmarshalFunc
+  castToUnmarshalFunc(UnmarshalFuncWithSlice func)
+  {
+    return reinterpret_cast<UnmarshalFunc>(func);
+  }
+
```

Comment:
Perhaps it would be better to consolidate m_unmarshal_func and m_unmarshal_func_map into an std::variant and also add an UnmarshalFuncWithSlice variant.

https://github.com/apache/trafficserver/blob/12801c966e5710d818287e79a7a977e837bfddc6/proxy/logging/LogField.h#L192